library/core/src/mem/mod.rs

```diff
@@ -908,15 +908,9 @@ pub fn take<T: Default>(dest: &mut T) -> T {
 #[must_use = "if you don't need the old value, you can just assign the new value directly"]
 #[rustc_const_unstable(feature = "const_replace", issue = "83164")]
 #[cfg_attr(not(test), rustc_diagnostic_item = "mem_replace")]
-pub const fn replace<T>(dest: &mut T, src: T) -> T {
-    // SAFETY: We read from `dest` but directly write `src` into it afterwards,
-    // such that the old value is not duplicated. Nothing is dropped and
-    // nothing here can panic.
-    unsafe {
-        let result = ptr::read(dest);
-        ptr::write(dest, src);
-        result
-    }
+pub const fn replace<T>(dest: &mut T, mut src: T) -> T {
+    swap(dest, &mut src);
+    src
```

Comment:
You were essentially trying to revert https://github.com/rust-lang/rust/pull/83022 here.

Maybe we should have a comment explaining why this is not calling `swap`?

Reply:
I think a comment is a good idea. It's not immediately clear that `swap` was being avoided for performance reasons.

Reply:
Do you want to make a PR for that? :)